site/app/controllers/ApiTokenController.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace app\controllers;
+
+use app\libraries\response\RedirectResponse;
+use app\libraries\response\WebResponse;
+use app\libraries\routers\AccessControl;
+use app\libraries\TokenManager;
+use app\views\ApiTokenView;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * Class ApiTokenController
+ * @package app\controllers
+ *
+ * @AccessControl(level="FACULTY")
```

Comment:
The API is already limited to faculty only so I didn't really see a point of giving non-faculty access to the page with these tokens. If we ever decide to allow API endpoints that are for all users then we could easily just drop the access control.

I don't think this should be feature flagged so that faculty can start getting used to this first until it's required to use these. Also we can't feature flag a non-course scope I'm pretty sure since feature flags get stored inside the course json.

Reply:
Right the APIs can be handled separately but viewing the token page is faculty only as well - I thought students would eventually use this for SAML git auth or is the plan to make a code change to enable this to everyone in the future?

Yeah right now feature flagging is for courses only so probably not worth the effort to add global config support to it depending on the timeline of rolling saml auth out

Reply:
Git Auth Tokens are completely separate from these API tokens. Here is the PR for Git Auth Tokens: https://github.com/Submitty/Submitty/pull/7721. The Git Auth Tokens page would be open to everyone.